modules/related-posts/jetpack-related-posts.php

```diff
@@ -248,6 +253,127 @@ public function get_target_html_unsupported() {
 		return "\n\n<!-- Jetpack Related Posts is not supported in this context. -->\n\n";
 	}
 
+	/**
+	 * ===============
+	 * GUTENBERG BLOCK
+	 * ===============
+	 */
+
+	/**
+	 * Render the related posts markup.
+	 *
+	 * @param array $attributes Block attributes.
+	 * @return string
+	 */
+	public function render_block( $attributes ) {
+		$block_attributes = array(
+			'show_thumbnails' => isset( $attributes['displayThumbnails'] ) && $attributes['displayThumbnails'],
+			'show_date'       => isset( $attributes['displayDate'] ) ? (bool) $attributes['displayDate'] : true,
+			'show_context'    => isset( $attributes['displayContext'] ) && $attributes['displayContext'],
+			'layout'          => isset( $attributes['postLayout'] ) && 'list' === $attributes['postLayout'] ? $attributes['postLayout'] : 'grid',
+			'size'            => ! empty( $attributes['postsToShow'] ) ? absint( $attributes['postsToShow'] ) : 3,
+		);
+
+		$excludes = array();
+		if ( isset( $_GET['relatedposts_origin'] ) ) {
+			if ( is_string( $_GET['relatedposts_origin'] ) ) {
+				$excludes = explode( ',', $_GET['relatedposts_origin'] );
+			} elseif ( is_array( $_GET['relatedposts_origin'] ) ) {
+				$excludes = array_values( $_GET['relatedposts_origin'] );
+			}
+
+			$excludes = array_unique( array_filter( array_map( 'absint', $excludes ) ) );
+		}
+
+		$related_posts = $this->get_for_post_id(
+			get_the_ID(),
+			array(
+				'size'             => $block_attributes['size'],
+				'exclude_post_ids' => $excludes,
+			)
+		);
+
+		if ( ! $related_posts ) {
+			return '';
+		}
+
+		remove_filter( 'the_content', array( $this, 'filter_add_target_to_dom' ), 40 );
```

Comment:
Instead of using the filter here, I wonder if it would be more readable if we used the same method as for the shortcode:
https://github.com/Automattic/jetpack/blob/14cee83bcc4a545331151fedd02176afdbc78b27/modules/related-posts/jetpack-related-posts.php#L186

Reply:
Nice suggestion 👍 Done in ef92f22f765e756d8fde8ef5d3097c08728377d5.